src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -219,6 +244,90 @@ internal static void FreeSslContext(SafeSslHandle context)
         #endregion
 
         #region private methods
+
+        private static void QueryEndPointChannelBinding(SafeSslHandle context, SafeChannelBinding bindingHandle)
+        {
+            using (SafeX509Handle certSafeHandle = GetPeerCertificate(context))
+            {
+                if (certSafeHandle == null || certSafeHandle.IsInvalid)
+                {
+                    throw CreateSslException(SR.net_ssl_invalid_certificate);
+                }
+
+                int certHashLength = 0;
+                int retVal = Crypto.X509Digest(certSafeHandle, GetEvpMethodForChannelBinding(certSafeHandle), bindingHandle.CertHashPtr, ref certHashLength);
```

Comment:
Rather than calling X509_digest, why not just do the hashing in managed code?

```C#
using (X509Certificate2 cert = new X509Certificate2(certSafeHandle.DangerousGetHandle())
using (HashAlgorithm hash = GetHashForChannelBinding(cert))
{
    byte[] bindingHash = hash.ComputeHash(cert.RawData);
}
```
I'm fairly sure that's equivalent code, and doesn't add more IntPtr usages, or arbitrarily-sized data moving between native and managed. (And the other method would return HashAlgorithm instead of the IntPtr of the EVP_MD)